### PR TITLE
Define netapp_e_mgmt_interface storage module.

### PR DESCRIPTION
**Pull request by ndswartz**
##### SUMMARY
The netapp_e_mgmt_interface module provides management configuration for
e-series storage array interface points including remote ssh access, NTP
and DNS services.

This patch also includes integration and unit tests.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME
netapp_e_mgmt_interface

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (mgmt_testing_pub c09b1843c5) last updated 2018/07/09 10:33:34 (GMT -500)
  config file = None
  configured module search path = [u'/home/swartzn/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/swartzn/ansible/lib/ansible
  executable location = /home/swartzn/ansible/bin/ansible
  python version = 2.7.15rc1 (default, Apr 15 2018, 21:51:34) [GCC 7.3.0]

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Review by lmprice — APPROVED** [review]
shipit

File: lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py [hunk 1]
```diff
@@ -0,0 +1,708 @@
...
+            - A
+            - B
+    name:
+        description:
+            - The port to modify the configuration for.
+            - The list of choices is not necessarily comprehensive. It depends on the number of ports
+              that are present in the system.
+            - The name represents the port number (typically from left to right on the controller),
+              beginning with a value of 1.
+            - Mutually exclusive with I(channel).
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py:56]: You can add the following to `argument_spec`
`mutually_exclusive=(['name', 'channel'],),`

File: lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py [hunk 2]
```diff
@@ -0,0 +1,708 @@
...
+        aliases:
+            - port
+            - iface
+    channel:
+        description:
+            - The port to modify the configuration for.
+            - The channel represents the port number (typically from left to right on the controller),
+              beginning with a value of 1.
+            - Mutually exclusive with I(name).
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py:65]: Maybe worth adding code to check this and call `fail_json` if it isn't allowed

File: lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py [hunk 3]
```diff
@@ -0,0 +1,708 @@
...
+        description:
+            - Backup IPv4 NTP server address
+            - Queried when primary NTP server fails
+        required: no
+    ssh:
+        type: bool
+        description:
+            - Enable ssh access to the controller for debug purposes.
+            - This is a controller-level setting.
+            - Older controllers may enable rlogin or telnet instead of ssh.
```
Comment (gundalow) [inline lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py]: So this modules will do...?

**Review by gundalow — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py (hunk 3 above)
Comment (lmprice) [reply lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py]: This is a bit confusing if you're not familiar with the E-Series systems. This option will enable/disable telnet or ssh, depending on the age/model of the E-Series system that's being managed. Perhaps it's worded poorly.

Comment (gundalow) [reply lib/ansible/modules/storage/netapp/netapp_e_mgmt_interface.py]: Much better, thanks

**Pull request merged**